src/molecules/search-field/SearchField.tsx

```diff
@@ -55,16 +63,22 @@ const SearchField = forwardRef(
 		ref: ForwardedRef<HTMLDivElement>
 	) => {
 		const [displayResetButton, setDisplayResetButton] = useState(false);
+		const inputRef = useRef<HTMLInputElement>(null);
 
-		const inputChangeHandler: ChangeEventHandler = (e: ChangeEvent) => {
-			const element = e.target as HTMLInputElement;
+		useEffect(() => {
+			if (defaultValue) setDisplayResetButton(true);
+			else setDisplayResetButton(false);
+		}, [defaultValue]);
```

Comment:
I just thought of the fact that we really only need to run this effect once, when the component mounts, and the defaultValue prop is never supposed to change anyway.
```suggestion
		}, []);
```